brainglobe_utils/brainmapper/transform_widget.py

```diff
@@ -0,0 +1,780 @@
+            display_info(
+                self,
+                "No points layer selected",
+                "Please select a points layer you would like to transform",
+            )
+            return False
+
+        return True
+
+    def load_brainreg_directory(self) -> bool:
+        """
+        Load the brainreg directory selected by the user.
+        Returns false if not selected, to abort analysis.
+
+        Returns
+        -------
+        bool
+            True if a directory was selected, False otherwise.
+        """
+        brainreg_directory = QFileDialog.getExistingDirectory(
+            self,
+            "Select brainreg directory",
+        )
+        if brainreg_directory == "":
+            return False
+        else:
+            self.brainreg_directory = Path(brainreg_directory)
+
+        self.initialise_brainreg_data()
+        self.status_label.setText("Ready")
+        return True
+
+    def initialise_brainreg_data(self) -> None:
+        """
+        Initialize brainreg data by defining the paths,
+        then loading the brainreg metadata, and then the atlas.
+        """
+        self.get_brainreg_paths()
+        self.check_brainreg_directory()
+        self.get_registration_metadata()
+        self.load_atlas()
+
+    def get_brainreg_paths(self) -> None:
+        """
+        Get the relevant file paths from the brainreg
+        output directory chosen by the user
+        """
+        self.paths = Paths(self.brainreg_directory)
+
+    def check_brainreg_directory(self) -> None:
+        """
+        Check if the selected directory is a valid brainreg directory
+        by checking for the existence of an "atlas" entry in the json
+        """
+        try:
+            with open(self.paths.brainreg_metadata_file) as json_file:
+                self.brainreg_metadata = json.load(json_file)
+
+                if "atlas" not in self.brainreg_metadata:
+                    self.display_brainreg_directory_warning()
+
+        except FileNotFoundError:
+            self.display_brainreg_directory_warning()
+
+    def display_brainreg_directory_warning(self) -> None:
+        """
+        Display a warning to the user if the selected directory
+        is not a valid brainreg directory.
+        """
+        display_info(
+            self,
+            "Not a brainreg directory",
+            "This directory does not appear to be a valid brainreg "
+            "directory. Please try loading another brainreg output directory.",
+        )
+
+    def get_registration_metadata(self) -> None:
+        """
+        Load registration metadata (atlas, orientation, voxel sizes)
+        from the brainreg directory.
+        """
+        self.metadata = Metadata(self.brainreg_metadata)
+
+    def load_atlas(self) -> None:
+        """
+        Load the BrainGlobe atlas used for the initial brainreg registration.
+        """
+        if not self.is_atlas_installed(self.metadata.atlas_string):
```

Comment:
```suggestion
        is_atlas_installed = self.metadata.atlas_string in get_downloaded_atlases()
        if not is_atlas_installed:
```
would also avoid a level of abstraction that is maybe not needed (could remove `is_atlas_installed`)?

Reply:
I added this function because (IMO) it makes it easier to mock for tests. I lean towards keeping it in for now.